src/github.com/matrix-org/dendrite/roomserver/storage/storage.go

```diff
@@ -380,6 +382,27 @@ func (d *Database) StateEntriesForTuples(
 	return d.statements.bulkSelectFilteredStateBlockEntries(stateBlockNIDs, stateKeyTuples)
 }
 
+// MembershipUpdater implements input.RoomEventDatabase
+func (d *Database) MembershipUpdater(roomID, targetUserID string) (updater types.MembershipUpdater, err error) {
+	err = common.WithTransaction(d.db, func(txn *sql.Tx) error {
+		var (
+			roomNID       types.RoomNID
+			targetUserNID types.EventStateKeyNID
+		)
+		roomNID, err = d.assignRoomNID(txn, roomID)
```

Comment:
Umm, good question, I think because we already have a transaction lying around at this point and it's a bit of a waste to snaffle a free database connection to run the stmt outside the transaction.